### PR TITLE
add cyflann recipe

### PR DESCRIPTION
**Pull request by dougalsutherland**
[cyflann](https://github.com/dougalsutherland/cyflann) is my alternative interface to [flann](https://github.com/conda-forge/flann-feedstock/), rather than the default [pyflann](https://github.com/conda-forge/pyflann-feedstock/). It's not incredibly widely-used but I think it would be worthwhile to have in conda-forge.

### CODE REVIEW

**Comment by dougalsutherland** [comment]
I had a comment after the `skip: True  [win]` line in the initial submission, which I guess is why AppVeyor is running....

**Comment by synapticarbors** [comment]
LGTM -- Happy to merge assuming Travis builds without issue.

File: recipes/cyflann/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    - python
+    - setuptools
+    - toolchain
+    - numpy x.x
+    - flann >=1.9.1,<1.10
+    - cython
+  run:
+    - python
+    - numpy x.x
+    - flann >=1.9.1,<1.10
```
Comment (jakirkham) [inline recipes/cyflann/meta.yaml]: Not a blocker, but maybe it is time to add `flann` to the [pinnings]( https://github.com/conda-forge/conda-forge.github.io/blob/43e5e18d26341cb82c11dbcfba64f7958267a7f8/scripts/pin_the_slow_way.py#L40-L83 ). Would you be up for doing this @dougalsutherland?

Comment (dougalsutherland) [inline recipes/cyflann/meta.yaml]: Sure. My [shogun PR](https://github.com/conda-forge/staged-recipes/pull/3237) also needed to pin a bunch of other libraries: arpack, curl, glpk, json-c, lzo, protobuf, snappy (which is why I sent version bump PRs to a bunch of those today). Should I add those too?

Comment (jakirkham) [inline recipes/cyflann/meta.yaml]: `curl` has been really stable. So haven't worried about it, but could be worth revisiting.

Sure it is worth doing these as well. May want to break up the PRs, but it is up to you.

**Comment by dougalsutherland** [comment]
Hmm, something's up with the include path on Travis, though it seemed to build locally. Will investigate.

File: recipes/cyflann/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/cyflann/meta.yaml]: Thanks for all of those PRs. Let's see what others say.

**Comment by dougalsutherland** [comment]
Okay, this is working now; I forgot that it used pkg-config to look for flann. :)

File: recipes/cyflann/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+  skip:
+    # Windows should work, but flann recipe isn't built there yet.
+    - True  # [win]
```
Comment (jakirkham) [inline recipes/cyflann/meta.yaml]: Please make this

```yaml
  # Windows should work, but flann recipe isn't built there yet.
  skip: true  # [win]
```

Comment (dougalsutherland) [inline recipes/cyflann/meta.yaml]: Not sure why I wrote it that way. :)  Done.

Comment (jakirkham) [inline recipes/cyflann/meta.yaml]: No worries. Thanks.

**Comment by jakirkham** [comment]
Thanks @dougalsutherland.

**Pull request merged**